src/main/java/com/pingcap/tikv/util/ScalarFuncInfer.java

```diff
@@ -0,0 +1,85 @@
+package com.pingcap.tikv.util;
+
+import com.pingcap.tidb.tipb.ScalarFuncSig;
+import com.pingcap.tikv.exception.TypeException;
+import com.pingcap.tikv.types.*;
+
+import java.util.Objects;
+
+/**
+ * The ScalarFunction Signature inferrer.
+ * <p>
+ * Used to infer a target signature for the given DataType
+ */
+public class ScalarFuncInfer {
+  /**
+   * Infer scalar function signature.
+   * You should provide candidates for the
+   * inferrer to choose.
+   *
+   * @param dataType     the data type
+   * @param intSig       the int sig
+   * @param decimalSig   the decimal sig
+   * @param realSig      the real sig
+   * @param durationType the duration type
+   * @param timeType     the time type
+   * @return the scalar func sig
+   */
+  public static ScalarFuncSig infer(DataType dataType,
+                                    ScalarFuncSig intSig,
+                                    ScalarFuncSig decimalSig,
+                                    ScalarFuncSig realSig,
+                                    ScalarFuncSig durationType,
+                                    ScalarFuncSig timeType) {
+    Objects.requireNonNull(dataType, "Data type should not be null!");
+
+    if (dataType instanceof IntegerType) {
+      Objects.requireNonNull(intSig, "No IntegerType signature provided!");
```

Comment:
Actually you can do return requireNonNull(xxxx) and import static for Objects.requireNonNull. Looks concise.

Reply:
Thanks for your advice. Done.